Yep but it will break a bunch of packages. I do not have the resources to do a full `nixpkgs-review` run right now but a partial run shows a bunch of issues:

```
error: builder for '/nix/store/z5rkz9mpc3373mqg2ak2ixncfx91pssp-python3.10-poetry-1.2.2.drv' failed with exit code 1;
       last 10 log lines:
       > Requirement already satisfied: requests-toolbelt<0.10.0,>=0.9.1 in /nix/store/g6klr9lgfrkxf7wkdsf7r9pc39n45hgd-python3.10-requests-toolbelt-0.9.1/lib/python3.10/site-packages (from poetry==1.2.2) (0.9.1)
       > Requirement already satisfied: pexpect<5.0.0,>=4.7.0 in /nix/store/lxinahd9qhdzmqia3493mrgkqh3ingig-python3.10-pexpect-4.8.0/lib/python3.10/site-packages (from poetry==1.2.2) (4.8.0)
       > Requirement already satisfied: poetry-core==1.3.2 in /nix/store/y7m62x2v20cxga1vpqr7kmbgc6nibs5s-python3.10-poetry-core-1.3.2/lib/python3.10/site-packages (from poetry==1.2.2) (1.3.2)
       > Requirement already satisfied: shellingham<2.0,>=1.5 in /nix/store/2qskk48bga2hpwcy2bpp9idcx9bf370q-python3.10-shellingham-1.5.0/lib/python3.10/site-packages (from poetry==1.2.2) (1.5.0)
       > Requirement already satisfied: requests<3.0,>=2.18 in /nix/store/msn1w5hapjnckz4dnkjpmjwp88421a5z-python3.10-requests-2.28.1/lib/python3.10/site-packages (from poetry==1.2.2) (2.28.1)
       > Requirement already satisfied: jsonschema<5.0.0,>=4.10.0 in /nix/store/cxn5s0s9zgh0czm8w85wwi3rjpvnfk2c-python3.10-jsonschema-4.17.0/lib/python3.10/site-packages (from poetry==1.2.2) (4.17.0)
       > Requirement already satisfied: pkginfo<2.0,>=1.5 in /nix/store/qnnhxhsylg7m63blsy13wk86z6xivn17-python3.10-pkginfo-1.8.3/lib/python3.10/site-packages (from poetry==1.2.2) (1.8.3)
       > ERROR: Could not find a version that satisfies the requirement cleo<2.0.0,>=1.0.0a5 (from poetry) (from versions: none)
       > ERROR: No matching distribution found for cleo<2.0.0,>=1.0.0a5
       > 
       For full logs, run 'nix log /nix/store/z5rkz9mpc3373mqg2ak2ixncfx91pssp-python3.10-poetry-1.2.2.drv'.
error: 1 dependencies of derivation '/nix/store/44nbpzw1i0h7n23p6nxb412mlcf65pm6-cmake-language-server-0.1.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/x4kvablv4h9lcm2jrfqpvxzdrww9rjwj-commitizen-2.37.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hyzpbhllfslagkay8gnps8a74v7wirjr-poetry2conda-0.3.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hck7hrbv7kpa7fmj7cywlzql9mxdjjyk-python3.10-aioeafm-1.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/hw2x4kwa55kx4hqmfal0q9r45snw9849-python3.10-aioeafm-1.0.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/5i657qdzihs5gmvipx4apyigxccxx0fx-python3.10-aria2p-0.9.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/m727mvcmkgfvq2cfj40v0vrbrpnl2ycn-python3.10-expecttest-0.1.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a3mx88kfa3ry1jh6fphrh0agcnp34d6a-python3.10-hypothesis-auto-1.1.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/gr2vjq4cay7jifk516wkqlm8f1v0bjjm-python3.10-jedi-language-server-0.37.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/bgkivwx5jxz94ymdhdkgqwgvkx4wx3ng-python3.10-pipenv-poetry-migrate-0.2.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/b1agvxw454vkh3q69fc6kdkalanc38zz-python3.10-pyairnow-1.1.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/s12v2dbavb70wa1qm40sc3qynhdsdvh2-python3.10-pyairnow-1.1.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/45agyarfc41m9d4wsym7q320ynx1a7cj-python3.10-pytest-golden-0.2.2.drv' failed to build
error: 1 dependencies of derivation '/nix/store/v57q4d9ijmg8qx6qfbxnlcsfhxa197kw-python3.10-strawberry-graphql-0.125.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/8lazf3b1311s6xqi612a5ld3pl4fgchf-homeassistant-test-airnow-2022.11.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/3hamkasyvk015q6bnhacgv92nsgin0dr-homeassistant-test-eafm-2022.11.4.drv' failed to build
error: 1 dependencies of derivation '/nix/store/36bjqd4rrvwchpvi1rz2w0rjff5bqyv2-python3.10-functorch-0.2.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/a7i729ib816n2c1m9p4f8ayrzf9aqf2f-python3.10-ical-4.1.1.drv' failed to build
error: 1 dependencies of derivation '/nix/store/p11qvs4agxrl75dqn3fsfi0kfmf17qw7-streamdeck-ui-2.0.6.drv' failed to build
error: 1 dependencies of derivation '/nix/store/2adand7d4i7a4j1y424g69q5bkvswqn8-vscode-extension-ms-python-python-2022.17.13011006.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n184plx7npip0phbdqkk8zyxrxsay10f-ycmd-unstable-2022-08-15.drv' failed to build
error: 1 dependencies of derivation '/nix/store/aiv0y4wgwr9g203mi79mxyq62qsznyyp-python3.10-gcal-sync-4.0.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sc1lb9q9649dcb37pzxvsb53kpsih98n-python3.10-gcal-sync-4.0.3.drv' failed to build
error: 1 dependencies of derivation '/nix/store/xj4pz8947k9368kqgirqyxdf4knzsm76-vimplugin-YouCompleteMe-2022-11-19.drv' failed to build
error: 1 dependencies of derivation '/nix/store/sx1kdkfvayp0mmbgbzr83cakxsvvi49s-homeassistant-test-google-2022.11.4.drv' failed to build
error: builder for '/nix/store/5fn5whs8ipfdjn8nf01dgf8zgpg3kvy3-python3.10-pyatv-0.10.3.drv' failed with exit code 1;
       last 10 log lines:
       >   /build/source/tests/protocols/raop/test_raop_functional.py:484: DeprecationWarning: Call to deprecated function volume_down.
       >     await volume_interface.volume_down()
       >
       > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
       > =========================== short test summary info ============================
       > FAILED tests/protocols/mrp/test_mrp_auth.py::MrpAuthFunctionalTest::test_authentication
       > FAILED tests/protocols/mrp/test_mrp_auth.py::MrpAuthFunctionalTest::test_pairing_with_bad_pin
       > FAILED tests/protocols/mrp/test_mrp_auth.py::MrpAuthFunctionalTest::test_pairing_with_device
       > FAILED tests/protocols/mrp/test_mrp_auth.py::MrpAuthFunctionalTest::test_pairing_with_existing_credentials
       > =========== 4 failed, 995 passed, 1 skipped, 288 warnings in 11.97s ============
       For full logs, run 'nix log /nix/store/5fn5whs8ipfdjn8nf01dgf8zgpg3kvy3-python3.10-pyatv-0.10.3.drv'.
error: builder for '/nix/store/ianwyyi9m3shcx3y22127h676qrj1fb2-python3.9-poetry-1.2.2.drv' failed with exit code 1;
       last 10 log lines:
       > Successfully built poetry
       > Finished creating a wheel...
       > Finished executing pipBuildPhase
       > installing
       > Executing pipInstallPhase
       > /build/source/dist /build/source
       > Processing ./poetry-1.2.2-py3-none-any.whl
       > ERROR: Could not find a version that satisfies the requirement cleo<2.0.0,>=1.0.0a5 (from poetry) (from versions: none)
       > ERROR: No matching distribution found for cleo<2.0.0,>=1.0.0a5
       > 
       For full logs, run 'nix log /nix/store/ianwyyi9m3shcx3y22127h676qrj1fb2-python3.9-poetry-1.2.2.drv'.
```

At least for the backports will probably need to apply the patch to limit the potential breakages.

--- a/pkgs/applications/misc/streamdeck-ui/default.nix
+++ b/pkgs/applications/misc/streamdeck-ui/default.nix
@@ -1,7 +1,7 @@
 { lib
 , python3Packages
 , fetchFromGitHub
-, poetry
+, fetchpatch
 , copyDesktopItems
 , wrapQtAppsHook
 , writeText
@@ -19,6 +19,14 @@ python3Packages.buildPythonApplication rec {
     rev = "v${version}";
     sha256 = "sha256-5dk+5oefg5R68kv038gsZ2p5ixmpj/vBLBp/V7Sdos8=";
   };
+
+  patches = [
+    (fetchpatch {
+      name = "use-poetry-core.patch";
+      url = "https://github.com/timothycrosley/streamdeck-ui/commit/e271656c1f47b1619d1b942e2ebb01ab2d6a68a9.patch";
+      hash = "sha256-wqYwX6eSqMnW6OG7wSprD62Dz818ayFduVrqW9E/ays=";
+    })
+  ];
 
   desktopItems = [ (makeDesktopItem {
     name = "streamdeck-ui";
@@ -50,7 +58,7 @@ python3Packages.buildPythonApplication rec {
   format = "pyproject";
 
   nativeBuildInputs = [
-    poetry
+    python3Packages.poetry-core
     copyDesktopItems
     wrapQtAppsHook
   ];

--- a/pkgs/development/python-modules/aioeafm/default.nix
+++ b/pkgs/development/python-modules/aioeafm/default.nix
@@ -2,7 +2,8 @@
 , aiohttp
 , buildPythonPackage
 , fetchFromGitHub
-, poetry
+, fetchpatch
+, poetry-core
 , pytest-aiohttp
 , pytest-asyncio
 , pytest-cov
@@ -21,7 +22,15 @@ buildPythonPackage rec {
     sha256 = "048cxn3fw2hynp27zlizq7k8ps67qq9sib1ddgirnxy5zc87vgkc";
   };
 
-  nativeBuildInputs = [ poetry ];
+  patches = [
+    (fetchpatch {
+      name = "use-poetry-core.patch";
+      url = "https://github.com/Jc2k/aioeafm/commit/549590e2ed465be40e2406416d89b8a8cd8c6185.patch";
+      hash = "sha256-cG/vQI1XQO8LVvWsHrAj8KlPGRulvO7Ny+k0CKUpPqQ=";
+    })
+  ];
+
+  nativeBuildInputs = [ poetry-core ];
 
   propagatedBuildInputs = [ aiohttp ];
 

--- a/pkgs/development/python-modules/hypothesis-auto/default.nix
+++ b/pkgs/development/python-modules/hypothesis-auto/default.nix
@@ -1,8 +1,9 @@
 { lib
 , buildPythonPackage
 , fetchPypi
+, fetchpatch
 , hypothesis
-, poetry
+, poetry-core
 , pydantic
 , pytest
 , pythonOlder
@@ -20,6 +21,14 @@ buildPythonPackage rec {
     hash = "sha256-XiwvsJ3AmEJRLYBjC7eSNZodM9LARzrUfuI9oL6eMrE=";
   };
 
+  patches = [
+    (fetchpatch {
+      name = "switch-to-poetry-core.patch";
+      url = "https://github.com/timothycrosley/hypothesis-auto/commit/8277b4232617c0433f80e9c2844452b9fae67a65.patch";
+      hash = "sha256-/0z0nphtQnUBiLYhhzLZT59kQgktSugaBg+ePNxy0qI=";
+    })
+  ];
+
   postPatch = ''
     # https://github.com/timothycrosley/hypothesis-auto/pull/20
     substituteInPlace pyproject.toml \
@@ -28,7 +37,7 @@ buildPythonPackage rec {
   '';
 
   nativeBuildInputs = [
-    poetry
+    poetry-core
   ];
 
   propagatedBuildInputs = [

